File: uc/computerScience2/fall2013/liveDemos/week03/pizza-04/topping.cpp
#include "topping.h"

Topping::Topping() {
  // private, doesn't do anything
}

Topping::Topping(string newName, bool isPremium) {
  name = newName;
  premium = isPremium;
}

string Topping::getName() {
  return name;
}

bool Topping::isPremium() {
  return premium;
}

File: uc/computerScience2/spring2014/liveDemos/java/pizza/src/edu/uc/eecs/pizza/Topping.java
package edu.uc.eecs.pizza;

public class Topping implements Comparable<Topping> {
  private String name;
  private boolean premium;
  
  Topping(String name, boolean isPremium) {
    this.name = name;
    this.premium = isPremium;
  }

  Topping(String name) {
    this(name, false);
  }
  
  String getName() {
    return name;
  }

  boolean isPremium() {
    return premium;
  }

  @Override
  public int compareTo(Topping that) {
    return this.name.compareTo(that.name);
  }
  
  @Override
  public String toString() {
    return name + " " + premium;
  }
}
